.github/workflows/unit_tests.yml

```diff
@@ -15,12 +11,11 @@ jobs:
 
     runs-on: ${{ matrix.os }}
     strategy:
-      max-parallel: 4
```

Comment:
Not really tied to this. `codecov` was a bit flaky receiving multiple reports simultaneously and often times it was erroring out. This was mitigating the issue while not adding a lot of overhead (tests still complete in a few minutes). Anyhow I don't have real data to back up this claim, so if it is working and you strongly prefer going parallel that's fine.

Reply:
"strongly" is a bit... strongly worded. I see no reason to restrict it to 4 (and if something pops up, we can add it back)

Reply:
Fine with me.